Bb 13 event card component

DylanHopkins21 opened the pull request:
## Describe your changes:
Created the EventCard component. Takes in all the props described in the Jira Issue ticket. Additionally, it takes in the owner boolean, displaying the owner badge if true. 

## Proof that your changes work (include screenshots if frontend):
<img width="438" alt="image" src="https://user-images.githubusercontent.com/112282141/227023681-b4821a14-a84c-4f45-ba4e-b4948742c5be.png">



## Jira Issue ticket number and link:
BB-13
https://bearbuddies.atlassian.net/jira/software/projects/BB/boards/1?selectedIssue=BB-13

## Checklist before requesting a review
- [*] I have performed a self-review of my code
- [*] I have DM'd the group chat that I have made a PR and need reviewers.
- [*] I haved moved the Jira issue to "Under Review"
- [*] I have run `npx eslint filename.js` on all JavaScript files I've changed in this PR and fixed all displayed errors

DylanHopkins21 committed:
first attempt at Card Bootstrap

DylanHopkins21 committed:
photo not rendering

DylanHopkins21 committed:
made header and body of cards

DylanHopkins21 committed:
finished hardcoded card

DylanHopkins21 committed:
Added Props for dynamic event cards

DylanHopkins21 committed:
Added propTypes

thomaspwang commented on src/components/EventCard/EventCard.js:
```diff
@@ -0,0 +1,95 @@
+        joined: Boolean,
+        owner: Boolean,
+    }
+    let buttText;
+    if (joined) {
+        buttText = "Happening Soon!";
+    } else {
+        buttText = "Join!"
+    }
+    let badge;
```
use const instead of let

thomaspwang replied:
use const

thomaspwang commented on src/components/EventCard/EventCard.js:
```diff
@@ -0,0 +1,95 @@
+        joined: Boolean,
+        owner: Boolean,
+    }
+    let buttText;
```
use const

thomaspwang commented on src/components/EventCard/EventCard.js:
```diff
@@ -0,0 +1,95 @@
+        description: String,
+        joined: Boolean,
+        owner: Boolean,
+    }
+    let buttText;
+    if (joined) {
+        buttText = "Happening Soon!";
+    } else {
+        buttText = "Join!"
```
as funny as it is name buttText to buttonText

DylanHopkins21 replied:
Fixed

thomaspwang commented on src/components/EventCard/EventCard.js:
```diff
@@ -0,0 +1,95 @@
+        description: String,
+        joined: Boolean,
+        owner: Boolean,
+    }
+    let buttText;
+    if (joined) {
+        buttText = "Happening Soon!";
+    } else {
+        buttText = "Join!"
+    }
+    let badge;
+    if (owner) {
+        badge = <OwnerBadge />
+    } else {
+        badge = <div />
+    }
```
add spaces between logic blocks so easier to read

- space between let buttText
- space before let badge

DylanHopkins21 replied:
Fixed

<img width="443" alt="image" src="https://user-images.githubusercontent.com/112282141/227413658-ac05c7f3-89c5-4b24-a919-ac447492927a.png">

thomaspwang commented on src/components/EventCard/EventCard.js:
for icon SVGs include those in the public folder and then import them in

DylanHopkins21 replied:
They are now all components that are imported in from the public folder

thomaspwang commented on src/components/EventCard/EventCard.js:
try to style it more similarly to the figma 
- image larger
- (around) equal whitespace at top / bottom of card
- slightly larger text and icons
- smaller capacity logo and text (should be the same size as the card text)
- smaller title probably

<img width="788" alt="image" src="https://user-images.githubusercontent.com/82493352/227068174-5833e60a-42fa-4daf-bbcb-944986751fd6.png">

DylanHopkins21 replied:
<img width="480" alt="image" src="https://user-images.githubusercontent.com/112282141/227413533-cb587f95-d205-4847-9257-43cfc9503759.png">

Made p much every change, I think the capacity stuff is close to the figma already

thomaspwang replied:
amazing, and looks great.

thomaspwang commented on src/components/EventCard/EventCard.js:
Make sure you're using your global styles for standard text, for example, the title and normal text.

currently it's not DM sans but it should be

DylanHopkins21 replied:
Made the change in globals.css, should now be solid for everyone's work from here on out

thomaspwang commented on src/components/EventCard/EventCard.js:
```diff
@@ -0,0 +1,95 @@
+    )
+}
+
+function OwnerBadge() {
+    return (
+        <div className={styles.iconText}>
+            <svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24">
+                <path fill="#7aa3ff" d="m12 23l-3-3H5q-.825 0-1.413-.588T3 18V4q0-.825.588-1.413T5 2h14q.825 0 1.413.588T21 4v14q0 .825-.588 1.413T19 20h-4l-3 3Zm0-11q1.45 0 2.475-1.025T15.5 8.5q0-1.45-1.025-2.475T12 5q-1.45 0-2.475 1.025T8.5 8.5q0 1.45 1.025 2.475T12 12Zm0 8.2l2.2-2.2H19v-1.15q-1.35-1.325-3.138-2.087T12 14q-2.075 0-3.863.763T5 16.85V18h4.8l2.2 2.2Z"/>
+            </svg>
+            <p id={styles.badgeText}>Your Group</p>
+        </div>
+    )
+}
```
generally it's syntactic practice to have secondary components above the export default component, so move it up

DylanHopkins21 replied:
<img width="490" alt="image" src="https://user-images.githubusercontent.com/112282141/227413942-7e887754-0549-4b83-9968-54c45ce4a215.png">

Moved

thomaspwang commented on src/styles/globals.css:
are you sure you should include this file in this PR? I thought this is in another PR.

DylanHopkins21 replied:
Included because I needed to fix the DM Sans that will affect everyone

thomaspwang reviewed (changes requested):
Looks great! Just a few minor changes and it should be pog

DylanHopkins21 committed:
Made PR Changes

DylanHopkins21 committed:
Merge branch 'main' into BB-13-Event-Card-Component

thomaspwang reviewed (approved):
Woo! Looks great to me. Thanks for clarifying your changes by responding to comments; they're super helpful when reviewing,

LeonJia1 reviewed (approved):
looks good to me

LeonJia1 commented:
also reorder ur imports >:(

LeonJia1 commented on src/components/EventCard/EventCard.js:
```diff
@@ -0,0 +1,93 @@
+import 'bootstrap/dist/css/bootstrap.min.css';
+import Badge from '../../../public/badge';
+import Clock from '../../../public/clock';
+import Location from '../../../public/location';
+import Paragraph from '../../../public/paragraph';
+import Person from '../../../public/person';
+import styles from './EventCard.module.css';
+import { Button } from 'react-bootstrap';
+import React from "react";
+import Image from 'next/image'
```
organize imports!

thomaspwang committed:
Re-ordered imports

also better practice is to do import Button from "react-bootstrap/Button" instead of import { Button } from "react-bootstrap"
